Java/core/src/main/java/com/amazon/randomcutforest/RandomCutForest.java

```diff
@@ -1023,21 +1022,25 @@ public DensityOutput getSimpleDensity(double[] point) {
      *                              values in the point. The length of the array
      *                              should be greater than or equal to the number of
      *                              missing values.
+     * @param centrality            a parameter that provides a central estimation
+     *                              versus a more random estimation
      * @return A point with the missing values imputed.
      */
-    public ArrayList<double[]> getConditionalCentralField(double[] point, int numberOfMissingValues,
-            int[] missingIndexes) {
+    public ArrayList<double[]> getConditionalField(double[] point, int numberOfMissingValues, int[] missingIndexes,
+            double centrality) {
         checkArgument(numberOfMissingValues > 0, "numberOfMissingValues must be greater than 0");
         checkNotNull(missingIndexes, "missingIndexes must not be null");
         checkArgument(numberOfMissingValues <= missingIndexes.length,
                 "numberOfMissingValues must be less than or equal to missingIndexes.length");
+        checkArgument(centrality >= 0 && centrality <= 1, "centrality needs to be in range [0,1]");
```

Comment:
A goal/tenent of the Java RCF2.0 version is to set up a framework that is futuristic and extensible. So that there can be multiple improvements in a continuous fashion without having a large version upgrade. So from that spirit, the question flows down to "what features should we have so that future changes would not be blocked". The biggest blocker in this context, if of course the serialized state, and the second biggest blocker is the API used for specific functions. If we do see "obvious next asks" (which may take some time to post, and perhaps await an user based on effort to impact ratio) then we should have that. So that's a long winded way of saying "specific use of the parameter would follow in subsequent PRs, based on need/ask". In terms of likelihood based interpretation, the current imputeVisitor does spell that out without changing the functionality.

Reply:
Got it. Thanks for the clarification.